packages/light-apps/src/IdentityHeader/IdentityHeader.tsx

```diff
@@ -216,6 +229,32 @@ export function IdentityHeader (props: Props) {
     );
   };
 
+  const isValidUrl = (apiUrl: string): boolean => {
+    return (
+      (apiUrl.length > 5) &&
+      // check that it starts with a valid ws identifier
+      (apiUrl.startsWith('ws://') || apiUrl.startsWith('wss://'))
+    );
+  };
+
+  const onSelectNode = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+    const url = data.value! as string;
+    if (isValidUrl(url)) {
+      const settings = Object.assign(uiSettings.get(), {
```

Comment:
You could just `uiSettings.set({ apiUrl: url })`, uiSettings takes care of the rest